### PR TITLE
2822 collect ubi and claim data

### PR DESCRIPTION
**Pull request by omerzam**
# Description

Collect UBI and claim data for dashboard

About # ([2822](https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/2822))

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: fuse/src/mapping.ts [hunk 1]
```diff
@@ -209,3 +199,92 @@ export function handleWhitelistedRemoved(event: WhitelistedRemoved): void {
...
+    dailyUbi.totalUBIDistributed = dailyUbi.totalUBIDistributed.plus(event.params.amount)
+  }
+
+  if (dailyUbi.totalClaims == null) {
+    dailyUbi.totalClaims = BigInt.fromI32(1)
+  } else {
+    dailyUbi.totalClaims = dailyUbi.totalClaims.plus(BigInt.fromI32(1))
+  }
+
+  let statistics = Statistics.load('statistics')
```
Comment (sirpy) [inline fuse/src/mapping.ts]: get this value from when you update the statistics record in aggregateStatisticsFromUBIClaimed. so we dont load it twice.

File: fuse/src/mapping.ts [hunk 2]
```diff
@@ -209,3 +199,92 @@ export function handleWhitelistedRemoved(event: WhitelistedRemoved): void {
...
+    statistics.totalClaims = BigInt.fromI32(1)
+  } else {
+    statistics.totalClaims = statistics.totalClaims.plus(BigInt.fromI32(1))
+  }
+
+  statistics.save()
+
+}
+
+function aggregateCitizenFromUBIClaimed(event: UBIClaimed, citizen: Citizen | null): void {
```
Comment (sirpy) [inline fuse/src/mapping.ts:252]: make sure you merge code review from previous PR correctly as you refactored the code in this PR

**Review by sirpy — CHANGES_REQUESTED** [review]

File: fuse/src/mapping.ts (hunk 1 above)
Comment (omerzam) [reply fuse/src/mapping.ts]: done

**Comment by sirpy** [comment]
read deployments also from stakingModel and upgradables

**Pull request merged**